admin/server.go

```diff
@@ -0,0 +1,63 @@
+package admin
+
+import (
+	"context"
+	"errors"
+
+	pb "github.com/onflow/flow-go/admin/admin"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type adminServer struct {
+	pb.UnimplementedAdminServer
+	commandQ chan<- *CommandRequest
+}
+
+type CommandRequest struct {
+	ctx          context.Context
+	command      string
+	data         map[string]interface{}
+	responseChan chan<- *CommandResponse
+}
+
+type CommandResponse struct {
+	err error
+}
+
+func (s *adminServer) RunCommand(ctx context.Context, in *pb.RunCommandRequest) (*pb.RunCommandResponse, error) {
```

Comment:
trying to undertand this better- Why don't we report error as part of the `RunCommandResponse` instead of expecting it in the `responseChan`?

Reply:
We return errors directly from the `RunCommand` function. gRPC then takes care of showing that error to the user.

`RunCommandResponse` is only for successful requests.